README.md

```diff
@@ -1,9 +1,220 @@
 # KubeSecondaryDNS
 DNS for KubeVirt VirtualMachines secondary interfaces
 
+## Prerequisites
+1. The KubeSecondaryDNS Deployment should be reachable from outside the cluster.
+2. The guest must have a guest agent installed, so the IPs of the secondary interfaces will be reported.
+3. Kubevirt must be installed, else the plugin would exit with error after 2 minutes, due to
+the informer logic. It will be recreated according k8s logic.
+4. If necessary, establish connectivity to cluster public IP via a relevant DNS entity that is used to
+reach the authoritative KubeSecondaryDNS server, such as DNSResolver, TLD NameServer, etc.
+
+When running KubeSecondaryDNS on top on kubevirtci, we achieve [1] as follows:
+1. Create a NodePort for the Deployment, use `31111` as the port.
+2. Kubevirtci forwards all traffic from host `127.0.0.1:31111` to the kubevirtci container.
+3. Traffic in the container, on port `31111` is forwarded to the control plane node of the cluster.
+
+This way `nslookup -port=31111 <FQDN> 127.0.0.1` from outside the cluster can reach the secondary-dns container.
+
+## Deployment
+Choose the desired release from  
+https://github.com/kubevirt/kubesecondarydns/releases
+and apply its manifest.  
+For example:
+```shell
+KSD_RELEASE=v0.0.5
+URL=https://github.com/kubevirt/kubesecondarydns/releases/download/${KSD_RELEASE}/secondarydns.yaml
+kubectl apply -f ${URL}
+```
+
+To undeploy it, use:
+```shell
+kubectl delete -f ${URL}
+```
+
+## Parameters
+You can update those parameters in the manifest.
+
+`DOMAIN` (default: `""`) - Determines the suffix of the FQDN.  
+Can be used in order to differentiate between different clusters (by including the cluster name as part of it for example).  
+The FQDN would look as follows:  
+If `DOMAIN` is not empty `<interface_name>.<vm_name>.<namespace>.vm.<DOMAIN>`  
+If `DOMAIN` is empty `<interface_name>.<vm_name>.<namespace>.vm`  
+
+`NAME_SERVER_IP` (default: `""`) - TODO
+
+## Development
+
+### Main operations
+```shell
+# run unit tests
+make test
+
+# bring up a local cluster with Kubernetes
+make cluster-up
+
+# build and deploy KubeSecondaryDNS
+make cluster-sync
+
+# run e2e tests on the cluster, requires cluster with installed operator,
+make functest
+
+# clean up all resources created by the operator from the cluster
+make cluster-clean
+
+# delete the cluster
+make cluster-down
+```
+
+### Cluster access
+```shell
+# access kubernetes API on the cluster
+./cluster/kubectl.sh get nodes
+
+# ssh into the cluster's node
+./cluster/cli.sh ssh node01
+```
+
+### Additional operations (done by cluster-sync)
+```shell
+# build image
+make build
+
+# push the image to the registry
+make push
+
+# deploy KubeSecondaryDNS manifest
+make deploy
+
+# create the node port
+make create-nodeport
+```
+
+### Code related operations
+```shell
+# perform auto-formatting on the source code (if not done by your IDE)
+make fmt
+
+# run go vet
+make vet
```

Comment:
`make vendor` is missing

Reply:
just selected few that looked more widely used, we don't include `make vendor` usually but i can add it